File: frontend/src/features/items/lib/itemAutomation.js
export const AUTOMATION_STATUSES = [
  { value: 'unreviewed', label: 'Не оценено', hint: 'Поддержка механик этой записи ещё не проверена.', tone: 'muted' },
  { value: 'full', label: 'Полностью', hint: 'Все применимые механики поддержаны сайтом. Выбор игрока и предусмотренное правилами ручное подтверждение могут оставаться.', tone: 'success' },
  { value: 'partial', label: 'Частично', hint: 'Сайт поддерживает часть механик. Остальные нужно учитывать вручную.', tone: 'warning' },
  { value: 'none', label: 'Не автоматизировано', hint: 'Запись содержит правила, но сайт пока не применяет их механику.', tone: 'muted' },
  { value: 'not_applicable', label: 'Не требуется', hint: 'У записи нет механик, требующих автоматизации.', tone: 'muted' },
]
export const PLAYER_INTERACTION_HINT = 'Применение затрагивает персонажа другого игрока и требует согласования или действий на его листе. Это отдельный признак, а не степень автоматизации.'
export function automationStatus(value) {
  return AUTOMATION_STATUSES.find(row => row.value === value) || AUTOMATION_STATUSES[0]
}
export function itemAutomationDraft(item) {
  return {
    automationStatus: automationStatus(item?.automationStatus).value,
    automationNote: item?.automationNote || '',
    requiresPlayerInteraction: !!item?.requiresPlayerInteraction,
  }
}

File: internal/store/item_automation.go
package store

import (
	"errors"
	"strings"
	"unicode/utf8"
)

// ItemAutomation is catalogue metadata shared by all item types, never instance rules.
type ItemAutomation struct {
	AutomationStatus          string `json:"automationStatus"`
	AutomationNote            string `json:"automationNote"`
	RequiresPlayerInteraction bool   `json:"requiresPlayerInteraction"`
}

// ItemMetadataPatch preserves omitted values when updating through HTTP or MCP.
type ItemMetadataPatch struct {
	ParentID                  *int64               `json:"-"` // MCP: negative removes parent, nil preserves it.
	DerivedFromItemID         *int64               `json:"derivedFromItemId"`
	DerivationKind            *string              `json:"derivationKind"`
	Compatibility             *[]ItemCompatibility `json:"compatibility"`
	Hidden                    *bool                `json:"hidden"`
	AutomationStatus          *string              `json:"automationStatus"`
	AutomationNote            *string              `json:"automationNote"`
	RequiresPlayerInteraction *bool                `json:"requiresPlayerInteraction"`
}

func (p *ItemMetadataPatch) Validate() error {
	if (p.DerivedFromItemID == nil) != (p.DerivationKind == nil) {
		return errors.New("Исходная запись и тип варианта указываются вместе")
	}
	if p.Compatibility != nil {
		if err := ValidateItemCompatibility(*p.Compatibility); err != nil {
			return err
		}
	}
	if p.AutomationStatus != nil {
		switch *p.AutomationStatus {
		case "unreviewed", "full", "partial", "none", "not_applicable":
		default:
			return errors.New("Неизвестный статус автоматизации")
		}
	}
	if p.AutomationNote != nil {
		note := strings.TrimSpace(*p.AutomationNote)
		if utf8.RuneCountInString(note) > 1000 {
			return errors.New("Комментарий об автоматизации: не более 1000 символов")
		}
		p.AutomationNote = &note
	}
	return nil
}

func (p ItemMetadataPatch) Initial() ItemAutomation {
	result := ItemAutomation{AutomationStatus: "unreviewed"}
	if p.AutomationStatus != nil {
		result.AutomationStatus = *p.AutomationStatus
	}
	if p.AutomationNote != nil {
		result.AutomationNote = *p.AutomationNote
	}
	if p.RequiresPlayerInteraction != nil {
		result.RequiresPlayerInteraction = *p.RequiresPlayerInteraction
	}
	return result
}
